src/api/tasks/TaskRouter.router.js

```diff
@@ -31,23 +37,53 @@ const TaskRouter = () => {
         throttle
       );
     } catch (e) {
+      logEmitter.emit(
+        "functionFail",
+        "TaskRouter.router",
+        "bulk/createtascomiregistration"
+      );
       await fail(406, res, e.message);
```

Comment:
These catches are for unexpected exceptions - we can rely on the error handler for these, but the error handler shouldnt handle returning a fail screen to the user - thats should be the router or the action depending on the fail. The router fails could for example however return a 500 and an error, 406 is probably not the best.

The actions can call await fail() or success() as appropriate - but the key thing is that they only handle expected situations and return screens to the user as necessary if that makes sense. E.g. so I said if it was a database connection fail - that would be handled above the action somewhere (in router in this case or by the framework), but if say - there was an exception because the email sender was busy - and we know sometimes this would happen - the action needs to handle it and call fail() as necessary.